File: resources/expected/ros2/isolette/lax/src/isolette_cpp_pkg/src/base_code/thermostat_regulate_temperature_manage_heat_source_mhst_base_src.cpp
#include "isolette_cpp_pkg/base_headers/thermostat_regulate_temperature_manage_heat_source_mhst_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

thermostat_regulate_temperature_manage_heat_source_mhst_base::thermostat_regulate_temperature_manage_heat_source_mhst_base() : Node("thermostat_regulate_temperature_manage_heat_source_mhst")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    thermostat_regulate_temperature_manage_heat_source_mhst_current_tempWstatus_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_current_tempWstatus",
        1,
        std::bind(&thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_current_tempWstatus, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_heat_source_mhst_lower_desired_temp_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::Tempimpl>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_lower_desired_temp",
        1,
        std::bind(&thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_lower_desired_temp, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_heat_source_mhst_upper_desired_temp_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::Tempimpl>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_upper_desired_temp",
        1,
        std::bind(&thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_upper_desired_temp, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_heat_source_mhst_regulator_mode_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::RegulatorMode>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_regulator_mode",
        1,
        std::bind(&thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_regulator_mode, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_heat_source_mhst_heat_control_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::OnOff>(
        "heat_source_cpi_heat_controller_heat_control",
        1);

    // timeTriggered callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&thermostat_regulate_temperature_manage_heat_source_mhst_base::timeTriggered, this), cb_group_);

}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::init_current_tempWstatus(isolette_cpp_pkg_interfaces::msg::TempWstatusimpl val) {
    current_tempWstatus_msg_holder = std::make_shared<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(val);
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::init_lower_desired_temp(isolette_cpp_pkg_interfaces::msg::Tempimpl val) {
    lower_desired_temp_msg_holder = std::make_shared<isolette_cpp_pkg_interfaces::msg::Tempimpl>(val);
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::init_upper_desired_temp(isolette_cpp_pkg_interfaces::msg::Tempimpl val) {
    upper_desired_temp_msg_holder = std::make_shared<isolette_cpp_pkg_interfaces::msg::Tempimpl>(val);
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::init_regulator_mode(isolette_cpp_pkg_interfaces::msg::RegulatorMode val) {
    regulator_mode_msg_holder = std::make_shared<isolette_cpp_pkg_interfaces::msg::RegulatorMode>(val);
}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_current_tempWstatus(const isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr msg)
{
    current_tempWstatus_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_lower_desired_temp(const isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr msg)
{
    lower_desired_temp_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_upper_desired_temp(const isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr msg)
{
    upper_desired_temp_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::handle_regulator_mode(const isolette_cpp_pkg_interfaces::msg::RegulatorMode::SharedPtr msg)
{
    regulator_mode_msg_holder = msg;
}

isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr thermostat_regulate_temperature_manage_heat_source_mhst_base::get_current_tempWstatus() {
    return current_tempWstatus_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr thermostat_regulate_temperature_manage_heat_source_mhst_base::get_lower_desired_temp() {
    return lower_desired_temp_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr thermostat_regulate_temperature_manage_heat_source_mhst_base::get_upper_desired_temp() {
    return upper_desired_temp_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::RegulatorMode::SharedPtr thermostat_regulate_temperature_manage_heat_source_mhst_base::get_regulator_mode() {
    return regulator_mode_msg_holder;
}

void thermostat_regulate_temperature_manage_heat_source_mhst_base::put_heat_control(isolette_cpp_pkg_interfaces::msg::OnOff msg)
{
    thermostat_regulate_temperature_manage_heat_source_mhst_heat_control_publisher_->publish(msg);
}

File: resources/expected/ros2/python-isolette/lax/src/isolette_py_pkg/isolette_py_pkg/base_code/thermostat_regulate_temperature_manage_heat_source_mhst_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from collections import deque
from isolette_py_pkg.user_code.thermostat_regulate_temperature_manage_heat_source_mhst_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import Tempimpl
from isolette_py_pkg_interfaces.msg import RegulatorMode
from isolette_py_pkg_interfaces.msg import OnOff

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class thermostat_regulate_temperature_manage_heat_source_mhst_base(Node):
    def __init__(self):
        super().__init__("thermostat_regulate_temperature_manage_heat_source_mhst")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.thermostat_regulate_temperature_manage_heat_source_mhst_current_tempWstatus_subscription_ = self.create_subscription(
            TempWstatusimpl,
            "thermostat_regulate_temperature_manage_heat_source_mhst_current_tempWstatus",
            self.handle_current_tempWstatus,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_heat_source_mhst_lower_desired_temp_subscription_ = self.create_subscription(
            Tempimpl,
            "thermostat_regulate_temperature_manage_heat_source_mhst_lower_desired_temp",
            self.handle_lower_desired_temp,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_heat_source_mhst_upper_desired_temp_subscription_ = self.create_subscription(
            Tempimpl,
            "thermostat_regulate_temperature_manage_heat_source_mhst_upper_desired_temp",
            self.handle_upper_desired_temp,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_heat_source_mhst_regulator_mode_subscription_ = self.create_subscription(
            RegulatorMode,
            "thermostat_regulate_temperature_manage_heat_source_mhst_regulator_mode",
            self.handle_regulator_mode,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_heat_source_mhst_heat_control_publisher_ = self.create_publisher(
            OnOff,
            "heat_source_cpi_heat_controller_heat_control",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1, self.timeTriggered, callback_group=self.cb_group_)

        self.current_tempWstatus_msg_holder = None
        self.lower_desired_temp_msg_holder = None
        self.upper_desired_temp_msg_holder = None
        self.regulator_mode_msg_holder = None

    def init_current_tempWstatus(self, val):
        self.current_tempWstatus_msg_holder = val

    def init_lower_desired_temp(self, val):
        self.lower_desired_temp_msg_holder = val

    def init_upper_desired_temp(self, val):
        self.upper_desired_temp_msg_holder = val

    def init_regulator_mode(self, val):
        self.regulator_mode_msg_holder = val

    def timeTriggered(self):
        raise NotImplementedError("Subclasses must implement this method")

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def handle_current_tempWstatus(self, msg):
        self.current_tempWstatus_msg_holder = msg

    def handle_lower_desired_temp(self, msg):
        self.lower_desired_temp_msg_holder = msg

    def handle_upper_desired_temp(self, msg):
        self.upper_desired_temp_msg_holder = msg

    def handle_regulator_mode(self, msg):
        self.regulator_mode_msg_holder = msg

    def get_current_tempWstatus(self):
        return self.current_tempWstatus_msg_holder

    def get_lower_desired_temp(self):
        return self.lower_desired_temp_msg_holder

    def get_upper_desired_temp(self):
        return self.upper_desired_temp_msg_holder

    def get_regulator_mode(self):
        return self.regulator_mode_msg_holder

    def put_heat_control(self, msg):
        self.thermostat_regulate_temperature_manage_heat_source_mhst_heat_control_publisher_.publish(msg)

